### PR TITLE
fix(cloud-logging): removing Logstash dependencies, using Cloud Logging Logback

### PR DESCRIPTION
**Pull request by Kef131**
## Description
Fixes b/362827197

Updates the Google Cloud Run manual logging sample for Java to use the officially recommended Google Cloud Logging Logback Appender (`google-cloud-logging-logback`) instead of the third-party `logstash-logback-encoder` plugin. 

Main changes:
* **Removed Logstash dependency:** Removed the `logstash-logback-encoder` plugin alongside custom Logstash `kv()` key-value parameters used to write JSON structured logs.
* **Adopted native Google Cloud Logging:** Switched to the `google-cloud-logging-logback` plugin, which natively supports structured logging and can output directly to standard output via `<redirectToStdout>true</redirectToStdout>`.
* **Updated Dependency Management:**
  * Added `libraries-bom` under `<dependencyManagement>` to manage Google Cloud library versions.
  * Replaced `logstash-logback-encoder` with `google-cloud-logging-logback`.
  * Kept `logback-classic` and `logback-core` aligned at version `1.4.14` to ensure seamless SLF4J 2.x provider binding.
* **Updated Logback Configuration:** Replaced the Logstash appender and custom JSON encoder field filters with Google's official `LoggingAppender`, configured with `<redirectToStdout>true</redirectToStdout>`.
* **Adjusted Tests:**
  * Adjusted the test initialization so that redirection to standard output happens in `@BeforeClass` before `spark-java` loads the class-level static logger.
  * Updated test assertions to check for `"INFO"` instead of `"NOTICE"`.

## Checklist
- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (None)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (None)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [x] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: run/logging-manual/src/main/java/com/example/cloudrun/App.java [hunk 1]
```diff
@@ -41,30 +41,27 @@ public static void main(String[] args) {
...
           // -- End log correlation code --
 
-          // Create a structured log entry using key value pairs.
+          // Create a structured log entry using MDC (Mapped Diagnostic Context) keys.
           // For instantiating the "logger" variable, see
           // https://cloud.google.com/run/docs/logging#run_manual_logging-java
-          logger.error(
-              "This is the default display field.",
-              kv("component", "arbitrary-property"),
-              kv("severity", "NOTICE"),
```
Comment (Kef131) [inline run/logging-manual/src/main/java/com/example/cloudrun/App.java:66]: Important to mention that the severity on SLF4J logger object is given by the logged method (e.g., logger.info() maps to INFO, logger.warn() to WARNING, logger.error() to ERROR).

**Review by iennae — APPROVED** [review]
